lockup/README.md

```diff
@@ -1,43 +1,64 @@
 # Lockup / Vesting contract
 
-# PLEASE WAIT UNTIL [ISSUE 141](https://github.com/near/core-contracts/issues/141) IS RESOLVED BEFORE DEPLOYING THE CONTRACT
-
 ## Overview
 
-This contract acts as an escrow that locks and holds an owner's tokens for a lockup period. A lockup period starts
-from the moment transfers are enabled by voting and lasts for the specified duration.
+This contract acts as an escrow that locks and holds an owner's tokens for a lockup period. 
+The contract contains of lockup and vesting processes that go simultaneously.
+
+A lockup period starts from the specified timestamp (or, if not specified, from moment when transfers are enabled by voting) 
+and lasts for the specified duration.
 It's also possible to lock tokens until the absolute timestamp. In this case the tokens will be available after the
 lockup period or after the timestamp, whichever is later.
 
-If transfers are not enabled yet, the contract keeps the account ID of the transfer poll contract.
-When the transfer poll is resolved, it returns the timestamp when it was resolved and it's used as the beginning of the
-lockup period.
+Vesting is an additional mechanism. It also locks the tokens, and it allows to configure 2 more options:
+1. Ability of terminating the contract from the Foundation.
+2. Cliff: the period when all tokens remain locked. After the cliff, there is a linear unlock till the end of the contract.
 
-Once all tokens are unlocked, everything is vested, and transfers are enabled, the owner can add a full access key to the
-account. This will allow the owner to turn this contract account into a regular account, claim the remaining tokens, and remove the contract
+Once transfers are enabled, all tokens are unlocked, everything is vested, and termination process is ended (if applicable), 
+the owner can add a full access key to the account. 
+This will allow the owner to turn this contract account into a regular account, claim the remaining tokens, and remove the contract
 or delete the account.
 
+### Lockup schedule
+
+Lockup is a mechanism of linear unlocking of tokens that could not be terminated.
+It is described by following fields:
+- `lockup_timestamp` - The moment when the tokens become to unlock (start of the contract).
+- `lockup_duration` - Alternative way to set the moment when the tokens become to unlock. 
+  The duration from the moment the transfers are enabled (`1602614338293769340`) to start of the contract.
+- `release_duration` - Length of the contract. `finish_timestamp = lockup_timestamp + release_duration`
+
 ### Vesting schedule
 
-The contract can contain a vesting schedule and serve as a vesting agreement between the NEAR Foundation (Foundation) and an employee (owner of contract).
+The contract can contain a vesting schedule and serve as a vesting agreement between the Foundation and an employee (owner of contract).
+
+Foundation is either the NEAR Foundation, or it could be third-party account. 
```

Comment:
if so, is `foundation` still the best name?

Reply:
@telezhnaya `foundation_account_id` is an unfortunate internal naming, but we still can use better wording on the documentation level and reference the internal naming explicitly pointing out where it comes from.